src/doc/rustc-dev-guide/src/building/bootstrapping/debugging-bootstrap.md

```diff
@@ -0,0 +1,40 @@
+# Debugging bootstrap
+
+> FIXME: this page could be expanded 
+
+## `tracing` in bootstrap
+
+To minimize bootstrap build times, `tracing` in bootstrap is gated behind the `tracing` cargo feature, which in turn is activated by `bootstrap.py` (via `x.py`) if `BOOTSTRAP_TRACING` env var is set.
+
+### Using `tracing` in bootstrap
+
+Due to the conditional compilation via the `tracing` cargo feature, this means that `tracing` usages in bootstrap need to be also gated behind the `tracing` feature:
+
+- `tracing` macros (like `trace!`) and anything from `tracing`, `tracing_subscriber` and `tracing-tree` will need to be gated by `#[cfg(feature = "tracing")]`.
+- `tracing`'s `#[instrument(..)]` macro will need to be gated like `#![cfg_attr(feature = "tracing", instrument(..))]`.
```

Comment:
I feel like these details don't need to be included here. Given the title "debugging bootstrap" I would expect content to be like "how to debug bootstrap invocations" and/or "how to enable tracing for bootstrap invocations". But instead, the page discusses why it's gated by the "tracing" feature and how to use it on the source code, which seems less relevant. Including a sample command with its output would likely be more useful.

Reply:
The reason I included this specifically is that I wanted to write down how this is wired together something (might not be immediately obvious), but I can see where you're coming from. I'll change the dev-guide content here to instead describe an example usage and example output.

Reply:
- I moved this remark about feature-gating into a comment next to `setup_tracing`.
- For r-d-g, I instead added example usage and example output as suggested.